tools/wheels/cibw_test_command.sh

```diff
@@ -15,17 +15,18 @@ fi
 if [[ $RUNNER_OS == "macOS"  && $RUNNER_ARCH == "X64" ]]; then
     # Not clear why this is needed but it seems on x86_64 this is not the default
     # and without it f2py tests fail
-    export DYLD_LIBRARY_PATH=/usr/local/lib
+    export DYLD_LIBRARY_PATH="$DYLD_LIBRARY_PATH:/usr/local/lib"
+    # Needed so gfortran (not clang) can find system libraries like libm (-lm)
+    # in f2py tests
+    export LIBRARY_PATH="$LIBRARY_PATH:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib"
 fi
 # Set available memory value to avoid OOM problems on aarch64.
 # See gh-22418.
 export NPY_AVAILABLE_MEM="4 GB"
 if [[ $(python -c "import sys; print(sys.implementation.name)") == "pypy" ]]; then
-  # make PyPy more verbose, try to catc a segfault in
-  # numpy/lib/tests/test_function_base.py
+  # make PyPy more verbose, try to catch a segfault
   python -c "import sys; import numpy; sys.exit(not numpy.test(label='full', verbose=2))"
```

Comment:
Could probably remove the verbosity increment, but PyPy is flaky enough that it makes sense to be defensive here.